ormar/relations/relation.py

```diff
@@ -130,13 +130,20 @@ def _find_existing(
             raise ValueError("Cannot find existing models in parent relation type")
         if self._to_remove:
             self._clean_related()
-        for ind, relation_child in enumerate(self.related_models[:]):
+
+        if child not in self.related_models:
+            return None
+        else:
+            # We need to clear the weakrefs that don't point to anything anymore
+            # There's an assumption here that if some of the related models went out of scope,
```

Comment:
I put a comment, but I just wanted to highlight it: before, every time it checked for related models, it would check every item in the list for weakrefs. That leads to N^2 performance since for every related model you add, you have to run through the whole list. To get around this, I changed it to basically only check the first item and see if it's a weakref. If it is, check the whole list--otherwise, continue on.

The assumption is that there won't be weakrefs that go out of scope randomly in the middle of the list. As far as I can tell, that's a good assumption, but I wanted to ask about it.